src/format/multifactor/FidoAuthenticationFactor.cpp

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (C) 2024 KeePassXC Team <team@keepassxc.org>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 or (at your option)
+ * version 3 of the License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "FidoAuthenticationFactor.h"
+
+FidoAuthenticationFactor::FidoAuthenticationFactor(const QSharedPointer<AuthenticationFactor>& factor)
```

Comment:
Generally speaking, QSharedPointer should not be passed by reference since that defeats the purpose of using a shared pointer. It doesn't really matter here since you are not actually sharing this pointer, but just a note.

Reply:
I've removed every reference-to-QSharedPointer I added. Now there are more pointer copies happening. I have not removed other sites in the code that also pass references to QSharedPointer, only those I added.

Reply:
Thanks, and yah other parts of our code does this and needs cleanup eventually